internal/api/external.go

```diff
@@ -146,6 +180,67 @@ func (a *API) internalExternalProviderCallback(w http.ResponseWriter, r *http.Re
 		providerAccessToken = oAuthResponseData.token
 		providerRefreshToken = oAuthResponseData.refreshToken
 	}
+	flowStateID := getFlowStateID(ctx)
+	// if there's a non-empty FlowStateID we perform PKCE Flow
+	if flowStateID != "" {
+		var rq url.Values
+		var authCode string
+		if err := r.ParseForm(); r.Method == http.MethodPost && err == nil {
+			rq = r.Form
+		} else {
+			rq = r.URL.Query()
+		}
```

Comment:
What if err is non-nil? I know it's unlikely but it will fail silently here if there's any problem with ParseForm(). Can you handle it?